src/disk_manager.rs

```diff
@@ -304,9 +304,9 @@ fn get_disks() -> BynarResult<Vec<Disk>> {
         let mut d = Disk::new();
         let dev_path = Path::new("/dev/").join(device.name);
         // This will skip partition_info if it fails to gather.  Blank disks will fail
-        let p = get_partition_info(&dev_path).unwrap_or(PartitionInfo::new());
+        let p = get_partition_info(&dev_path).unwrap_or_else(|_| PartitionInfo::new());
         //Translate block_utils MediaType -> Protobuf DiskType
-        d.set_field_type(convert_media_to_disk_type(device.media_type));
+        d.set_field_type(convert_media_to_disk_type(&device.media_type));
         d.set_dev_path(dev_path.to_string_lossy().into_owned());
         d.set_partitions(p);
```

Comment:
No it doesn't modify the disks.  It's just in the gathering phase here.  That just says unwrap the Result or set it to a default value instead of returning early with an error.